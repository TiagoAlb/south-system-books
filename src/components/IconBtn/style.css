.tooltip:hover button {
  color: #fc4f00;
}