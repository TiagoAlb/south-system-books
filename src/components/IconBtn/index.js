import React from 'react'
import Icon from '@material-ui/core/Icon'
import IconButton from '@material-ui/core/IconButton'
import Tooltip from '../Tooltip'
import './style.css'

const IconBtn = (props) => {
    return (
        <Tooltip text={props.title}>
            <IconButton onClick={props.click}>
                <Icon>{props.icon}</Icon>
            </IconButton>
        </Tooltip>
    )
}

export default IconBtn