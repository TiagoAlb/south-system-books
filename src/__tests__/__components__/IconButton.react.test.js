import React from 'react'
import IconBtn from '../../components/IconButton'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom'

it('renders correctly', () => {
    const tree = renderer.create(
        <IconBtn
            title='Test'
            click={() => alert('Aqui teste')}
            icon='test' />).toJSON()
    expect(tree).toMatchSnapshot()
})