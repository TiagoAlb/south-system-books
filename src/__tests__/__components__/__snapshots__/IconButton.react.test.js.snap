// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="tooltip"
>
  <button
    className="MuiButtonBase-root MuiIconButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <span
        aria-hidden={true}
        className="material-icons MuiIcon-root"
      >
        test
      </span>
    </span>
  </button>
  <span
    className="tooltip-text"
  >
    Test
  </span>
</div>
`;
